src/jdk.incubator.code/share/classes/jdk/incubator/code/op/CoreOp.java

```diff
@@ -3880,7 +3918,20 @@ public static NewOp _new(TypeElement returnType, FunctionType constructorType,
      */
     public static NewOp _new(TypeElement returnType, FunctionType constructorType,
                              List<Value> args) {
-        return new NewOp(returnType, constructorType, args);
+        return new NewOp(false, returnType, constructorType, args);
+    }
+
+    /**
+     * Creates an instance creation operation.
+     *
+     * @param returnType      the instance type
+     * @param constructorDescriptor the constructor descriptor
+     * @param args            the constructor arguments
+     * @return the instance creation operation
+     */
+    public static NewOp _new(boolean isVarargs, TypeElement returnType, ConstructorRef constructorDescriptor,
```

Comment:
Update all the factory methods to accept `ConstructorRef` instead of `FunctionType`.